e2e/testcases/profiling_test.go

```diff
@@ -0,0 +1,608 @@
+				fmt.Sprintf("%s/namespaces/%s/deployment-%s.yaml", syncPath, ns, deployName),
+				[]byte(fmt.Sprintf(`
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: %s
+  namespace: %s
+  labels:
+    app: %s
+    nomos-test: enabled
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: %s
+  template:
+    metadata:
+      labels:
+        app: %s
+    spec:
+      containers:
+      - name: pause
+        image: registry.k8s.io/pause:3.7
+        ports:
+        - containerPort: 80
+        resources:
+          limits:
+            cpu: 250m
+            memory: 256Mi
+          requests:
+            cpu: 250m
+            memory: 256Mi
+`, deployName, ns, deployName, deployName, deployName))))
+		}
+
+		nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush(fmt.Sprintf("Adding %d deployments each for %d RootSyncs", deployCount, syncCount)))
+
+		// Add child RootSync
+		rSyncRefs = append(rSyncRefs, rSyncRef{
+			GroupVersionKind: kinds.RootSyncV1Beta1(),
+			NamespacedName:   nomostest.RootSyncNN(syncName),
+			SyncPath:         syncPath,
+			CommitFunc:       commitFn,
+		})
+
+		// Validate that the resources sync without the reconciler running out of
+		// memory, getting OOMKilled, and crash looping.
+		nt.Must(watchForSyncedAndReconciled(nt, rSyncRefs))
+
+		nt.T.Logf("Verify the number of Deployment objects")
+		nt.Must(validateNumberOfObjectsEquals(nt, kinds.Deployment(), totalDeployCount,
+			client.MatchingLabels{metadata.ManagedByKey: metadata.ManagedByValue},
+			client.InNamespace(ns)))
+	}
+
+	nt.T.Log("Removing RootSyncs from Git")
+	for syncIndex := 1; syncIndex <= (syncsPerStep * steps); syncIndex++ {
+		syncName := fmt.Sprintf("sync-%d", syncIndex)
+		nt.Must(nt.RootRepos[configsync.RootSyncName].Remove(
+			fmt.Sprintf("%s/namespaces/%s/rootsync-%s.yaml", gitproviders.DefaultSyncDir, configsync.ControllerNamespace, syncName)))
+	}
+
+	nt.T.Logf("Removing test namespace: %s", ns)
+	nt.Must(nt.RootRepos[configsync.RootSyncName].Remove(fmt.Sprintf("%s/ns-%s.yaml", gitproviders.DefaultSyncDir, ns)))
+	nt.Must(nt.RootRepos[configsync.RootSyncName].CommitAndPush("Removing RootSyncs and test namespace"))
+
+	// Wait for root-sync to sync
+	nt.Must(nt.WatchForAllSyncs())
+}
+
+// TestProfilingResourcesByRootSyncCount applies 10,000 Deployments, across
+// 10 different RootSyncs, each managing 1000 Deployments.
+// This is useful for profiling the resource requests, usage, and limits
+// relative to the total number of objects on the cluster, with a single
+// resource type, in the same namespace.
+// Skipped by default, because it needs cluster autoscaling and a lot of quota.
+func TestProfilingResourcesByRootSyncCount(t *testing.T) {
+	nt := nomostest.New(t, nomostesting.Reconciliation1, ntopts.Unstructured,
+		ntopts.ProfilingTest,
+		ntopts.WithReconcileTimeout(configsync.DefaultReconcileTimeout))
+
+	nt.T.Log("Stop the CS webhook by removing the webhook configuration")
+	nomostest.StopWebhook(nt)
```

Comment:
Why does this test specifically need the webhook stopped?

Reply:
It doesn't. It was just a little faster. I can remove the disable.